weasyprint/draw.py

```diff
@@ -184,6 +184,15 @@ def draw_stacking_context(context, stacking_context, enable_hinting):
     # See http://www.w3.org/TR/CSS2/zindex.html
     with stacked(context):
         box = stacking_context.box
+
+        # apply the viewport_overflow to the html box, see #35
+        if box.element_tag == 'html' and (
```

Comment:
Shouldn’t we use `box.is_for_root_element` instead?

Reply:
Wasn't aware that there is a `box.is_for_root_element`. 

Though, I always wondered about the  `# No root element` fallback in `build_formatting_structure()`. Never ever managed to construct an HTML/element_tree without a 'html' root node.

Reply:
> Though, I always wondered about the `# No root element` fallback in `build_formatting_structure()`. Never ever managed to construct an HTML/element_tree without a 'html' root node.

Good question, I don’t know if html5lib can give a document with no root `html` tag. Coverage says that this code is tested, so I suppose that it’s useful :smile:.

By the way, it’s easier to have an `html` tag nested in another `html` tag, that’s another good reason to use `is_for_root_element`.

Reply:
> have an html tag nested in another html tag

I tried that, too. Various scaring nestings of `html` and `body`. Creating multiple and/or nested `BlockBox body` is possible. 
But always ended up with one single `BlockBox html` root element.

Reply:
:scream: 

https://github.com/Kozea/WeasyPrint/blob/f333cefc515b15b878b6555d150e0471456ec90e/weasyprint/tests/test_boxes.py#L1324-L1334